Add support for the "csharp-netcore" OpenAPI Generator and RestSharp "non Portable"

danny-gallagher opened the pull request:
Providing library for netcore/Restsharp support

danny-gallagher committed:
Adding RestSharpV2 package to support openapi generator netcore libs

danny-gallagher committed:
Add condition for serialization of body

jaaufauvre commented:
Can we add the new project to the solution file (`Mastercard.Developer.OAuth1Signer.sln`)?

![image](https://user-images.githubusercontent.com/3964455/109016668-4a6d5b80-76ae-11eb-8056-79d935de6c5a.png)

jaaufauvre commented on Mastercard.Developer.OAuth1Signer.RestSharp/Mastercard.Developer.OAuth1Signer.RestSharp.csproj:
```diff
@@ -11,7 +11,7 @@
     <PackageReleaseNotes>See: https://github.com/Mastercard/oauth1-signer-csharp/releases</PackageReleaseNotes>
     <RepositoryUrl>https://github.com/Mastercard/oauth1-signer-csharp</RepositoryUrl>
-    <Description>Zero dependency library for generating a Mastercard API compliant OAuth signature</Description>
+    <Description>RestSharp Portable extension for generating a Mastercard API compliant OAuth signature</Description>
```
👍

jaaufauvre commented on Mastercard.Developer.OAuth1Signer.RestSharpV2/Mastercard.Developer.OAuth1Signer.RestSharpV2.csproj:
```diff
@@ -0,0 +1,38 @@
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
+    <Version>1.0.0</Version>
```
For simplicity, and because we release and publish the packages simultaneously I would keep one single and same version for the 3 projects: `1.3.0`

https://github.com/Mastercard/oauth1-signer-csharp/releases

jaaufauvre committed:
Added "Portable" to comments of the lib supporting RestSharp Portable

jaaufauvre commented on Mastercard.Developer.OAuth1Signer.RestSharpV2/Signers/RestSharpSigner.cs:
```diff
@@ -0,0 +1,78 @@
+            {
+                parameterString
+                    .Append(parameterString.Length > 0 ? "&" : string.Empty)
+                    .Append(Uri.EscapeDataString(requestParameter.Name))
```
We have two possible null assignment to a non-nullable entity here. I'm not sure if `requestParameter.Name` and `requestParameter.Value` can happen to be null, but we could add a quick test to ignore the parameter in that case.

jaaufauvre reviewed (changes requested):
Hey @danny-gallagher, nice work.
I added a few comments to the PR (nothing major).
Thanks!

danny-gallagher committed:
- Updating package versions
- Adding new package to solution file
- Throw exception when parameter name & value are null, and add tests

danny-gallagher committed:
Merge remote-tracking branch 'origin/feature/netcore-support' into feature/netcore-support

jaaufauvre reviewed (approved):
👍
